modules/data.land/R/extract_soil_nc.R

```diff
@@ -205,35 +178,34 @@ extract_soil_gssurgo <- function(outdir, lat, lon, size=1, grid_size=3, grid_spa
       )
     ) %>%
     dplyr::filter(stats::complete.cases(.))
+  
   if(nrow(soilprop.new) == 0) {
     PEcAn.logger::logger.error("No valid soil properties after filtering")
     return(NULL)
   } 
-  if(!dir.exists(outdir)) dir.create(outdir, recursive = TRUE)
   
-  #converting it to list
+  if(!dir.exists(outdir)) dir.create(outdir, recursive = TRUE)
+
   soil.data.gssurgo <- list(
     fraction_of_sand_in_soil = soilprop.new$fraction_of_sand_in_soil,
     fraction_of_silt_in_soil = soilprop.new$fraction_of_silt_in_soil,
     fraction_of_clay_in_soil = soilprop.new$fraction_of_clay_in_soil,
     soil_depth = soilprop.new$soil_depth,
     soil_organic_carbon_stock = soilprop.new$soil_organic_carbon_stock
   )
-  #This ensures that I have at least one soil ensemble in case the modeling part failed
-  all.soil.ens <-c(all.soil.ens, list(soil.data.gssurgo))
   
+  all.soil.ens <- c(all.soil.ens, list(soil.data.gssurgo))
   
-  # What I do here is that I put soil data into depth classes and then model each class speparatly
-  #- see if we need to generate soil ensemble and add that to the list of all
+  # Generate modeled ensembles
   tryCatch({
-    # find the soil depth levels based on the depth argument 
-    # if soil profile is deeper than what is specified in the argument then I go as deep as the soil profile.
-    if (max(soilprop.new$soil_depth) > max(depths)) {
-      depths <- sort(c(depths, max(soilprop.new$soil_depth)))
+    # Adjust depth levels if needed
+    if (max(soilprop.new$soil_depth_bottom) > max(depths_cm)) {
+      depths_cm <- sort(c(depths_cm, max(soilprop.new$soil_depth)))
```

Comment:
I don't understand fully, but the code compares against the deepest bottom soil depth but appends the (soil_depth) should this be:

```suggestion
      depths_cm <- sort(c(depths_cm, max(soilprop.new$soil_depth_bottom)))
```
?

Reply:
this is not necessary here now -- since aggregation already CLIPS to `bottom_depth` and it always sets to user requested depth